src/antlr/GroovyLexer.g4

```diff
@@ -948,7 +948,7 @@ SL_COMMENT
 // Script-header comments.
 // The very first characters of the file may be "#!".  If so, ignore the first line.
 SH_COMMENT
-    :   '#!' { require(0 == this.tokenIndex, "Shebang comment should appear at the first line", -2, true); } ~[\r\n\uFFFF]* -> skip
+    : '#!' { require(0 == this.tokenIndex, "Shebang comment should appear at the first line", -2, true); } ~[\r\n\uFFFF]* LineTerminator ('#!' ~[\r\n\uFFFF]* LineTerminator)* -> skip
```

Comment:
`LineTerminator ('#!' ~[\r\n\uFFFF]* LineTerminator)*`
I am not sure if the tailing `LineTerminator` is needed, maybe we could simplify the above code as:
`(LineTerminator '#!' ~[\r\n\uFFFF]* )*`

Reply:
This change worked for antlr4 but I'm having trouble with antlr2.

```
SH_COMMENT
options {
    paraphrase="a script header";
}
    :   {getLine() == 1 && getColumn() == 1}?  "#!"
        (
            options {  greedy = true;  }:
            // '\uffff' means the EOF character.
            // This will fix the issue GROOVY-766 (infinite loop).
            ~('\n'|'\r'|'\uffff')
        )*
        (ONE_NL[true] "#!"
          (
              options {  greedy = true;  }:
              // '\uffff' means the EOF character.
              // This will fix the issue GROOVY-766 (infinite loop).
              ~('\n'|'\r'|'\uffff')
          )*
        )*
        { if (!whitespaceIncluded)  $setType(Token.SKIP); }
    ;
```

I'm getting this error in the test

```
org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
subprojects/parser-antlr4/src/test/resources/core/Comments_03.groovy: 4: expecting '#', found '/' @ line 4, column 1.
   /*
   ^

1 error
```

Reply:
@eric-milles ping ;-)

Reply:
I think for Antlr4 I would go with:
```
    : '#!' { require(this.tokenIndex == 0, "Shebang comment should appear at the first line", -2, true); } ~[\r\n\uFFFF]* (LineTerminator '#!' ~[\r\n\uFFFF]*)* -> skip
```

For Antlr2, maybe this?
```
SH_COMMENT
options {
    paraphrase="a script header";
}
    :   {getLine() == 1 && getColumn() == 1}?
        "#!" (options { greedy = true; }: ~('\n'|'\r'|'\uffff'))*
        (ONE_NL "#!" (options { greedy = true; }: ~('\n'|'\r'|'\uffff'))*)*
        { if (!whitespaceIncluded) $setType(Token.SKIP); }
    ;
```